### PR TITLE
telegram command: advanced status

### PR DESCRIPTION
**Pull request by flightcom**
This adds commands for Telegram:
- `/count` will allow to see number of active trades
- options added to existing `/status` command
  + `table` outputs trades information in a table format
  + `table short` outputs a shortened version

### CODE REVIEW

**Comment by gcarq** [comment]
Hi, thanks for the PR

File: freqtrade/rpc/telegram.py [hunk 1]
```diff
@@ -146,6 +157,71 @@ def _status(bot: Bot, update: Update) -> None:
...
+                current_rate,
+                fmt_close_profit,
+                fmt_current_profit,
+                '{} ({})'.format(order['remaining'], order['type']) if order else 'No'
+            ]
+
+            trades_list.append(row)
+
+        columns = ['ID', 'Pair', 'Since', 'Amount', 'Open Rate',
+            'Close Rate', 'Cur. Rate', 'Close Profit', 'Profit',
```
Comment (gcarq) [inline freqtrade/rpc/telegram.py]: Fix indentation. (see pylint)

File: freqtrade/rpc/telegram.py [hunk 2]
```diff
@@ -146,6 +157,71 @@ def _status(bot: Bot, update: Update) -> None:
...
+            'Open Order']
+
+        df = DataFrame.from_records(trades_list, columns=columns)
+        df = df.set_index(columns[0])
+
+        columns_short = ['Pair', 'Since', 'Profit']
+
+        if short_version == True:
```
Comment (gcarq) [inline freqtrade/rpc/telegram.py]: this can be simplified to `if short_version:`

File: freqtrade/rpc/telegram.py [hunk 3]
```diff
@@ -331,6 +407,29 @@ def _performance(bot: Bot, update: Update) -> None:
...
+    :param update: message update
+    :return: None
+    """
+    if get_state() != State.RUNNING:
+        send_msg('`trader is not running`', bot=bot)
+        return
+
+    trades = Trade.query.filter(Trade.is_open.is_(True)).all()
+
+    current_trades_count = len(trades)
```
Comment (gcarq) [inline freqtrade/rpc/telegram.py]: There is no need do declare these variables, just pass the values to format(). You can also use placeholders to make it more explicit what is being passed.

File: freqtrade/rpc/telegram.py [hunk 4]
```diff
@@ -146,6 +157,71 @@ def _status(bot: Bot, update: Update) -> None:
...
+            orders = [o for o in orders if o['id'] == trade.open_order_id]
+            order = orders[0] if orders else None
+
+            fmt_close_profit = '{:.2f}'.format(trade.close_profit) if trade.close_profit else 'No'
+            fmt_current_profit = '{:.2f}'.format(current_profit)
+
+            row = [
+                trade.id,
+                trade.pair,
+                shorten_date(arrow.get(trade.open_date).humanize()),
```
Comment (gcarq) [inline freqtrade/rpc/telegram.py]: I think the `shorten_date` function can be simplified if you pass `only_distance=True` to humanize()

File: freqtrade/rpc/telegram.py [hunk 5]
```diff
@@ -146,6 +157,71 @@ def _status(bot: Bot, update: Update) -> None:
 
 
 @authorized_only
+def _status_table(bot: Bot, update: Update) -> None:
```
Comment (gcarq) [inline freqtrade/rpc/telegram.py]: R:160, 0: Too many local variables (18/15) (too-many-locals) (See pylint)

**Comment by gcarq** [comment]
@flightcom I really like the table short format.
I'm thinking about changing the default `/status` to `status table short` when this is merged, but I guess the width should reduced a little bit, because the last column is breaking (at least on the android client).

The normal table is not really helpful (at least for Telegram) because it is very hard to read on desktop and smartphone.

I left some comments, please also run pylint over the changed files.
A unittest would be also great, you can see how it is done in `tests/test_telegram.py` (not required for merging).

cheers

**Comment by flightcom** [comment]
Hi @gcarq, thank you for your comments. I'm new in python and did not have pylint installed.
I changed the normal output to be the `short` one. Plus, I added unit tests for new commands.
Let me know if there is anything else that requires some changes.

Cheers!

**Pull request merged**